substrate/frame/support/procedural/src/pallet/expand/view_functions.rs

```diff
@@ -0,0 +1,240 @@
+
+	let view_fn_impls = view_fns
+		.iter()
+		.map(|view_fn| expand_view_function(def, span, where_clause.as_ref(), view_fn));
+	let impl_dispatch_query = impl_dispatch_query(def, span, where_clause.as_ref(), &view_fns);
+	let impl_query_metadata =
+		impl_query_metadata(def, span, where_clause.as_ref(), &view_fns, &docs);
+
+	quote::quote! {
+		#query_prefix_impl
+		#( #view_fn_impls )*
+		#impl_dispatch_query
+		#impl_query_metadata
+	}
+}
+
+fn expand_query_prefix_impl(
+	def: &Def,
+	span: Span,
+	where_clause: Option<&syn::WhereClause>,
+) -> TokenStream {
+	let pallet_ident = &def.pallet_struct.pallet;
+	let frame_support = &def.frame_support;
+	let frame_system = &def.frame_system;
+	let type_impl_gen = &def.type_impl_generics(span);
+	let type_use_gen = &def.type_use_generics(span);
+
+	quote::quote! {
+		impl<#type_impl_gen> #frame_support::traits::QueryIdPrefix for #pallet_ident<#type_use_gen> #where_clause {
+			fn prefix() -> [::core::primitive::u8; 16usize] {
+				<
+					<T as #frame_system::Config>::PalletInfo
+					as #frame_support::traits::PalletInfo
+				>::name_hash::<Pallet<#type_use_gen>>()
+					.expect("No name_hash found for the pallet in the runtime! This usually means that the pallet wasn't added to `construct_runtime!`.")
+			}
+		}
+	}
+}
+
+fn expand_view_function(
+	def: &Def,
+	span: Span,
+	where_clause: Option<&syn::WhereClause>,
+	view_fn: &ViewFunctionDef,
+) -> TokenStream {
+	let frame_support = &def.frame_support;
+	let pallet_ident = &def.pallet_struct.pallet;
+	let type_impl_gen = &def.type_impl_generics(span);
+	let type_decl_bounded_gen = &def.type_decl_bounded_generics(span);
+	let type_use_gen = &def.type_use_generics(span);
+	let capture_docs = if cfg!(feature = "no-metadata-docs") { "never" } else { "always" };
+
+	let query_struct_ident = view_fn.query_struct_ident();
+	let view_fn_name = &view_fn.name;
+	let (arg_names, arg_types) = view_fn.args_names_types();
+	let return_type = &view_fn.return_type;
+	let docs = &view_fn.docs;
+
+	let query_id_suffix_bytes = view_fn
+		.query_id_suffix_bytes()
+		.map(|byte| syn::LitInt::new(&format!("0x{:X}_u8", byte), Span::call_site()));
+
+	quote::quote! {
+		#( #[doc = #docs] )*
+		#[allow(missing_docs)]
+		#[derive(
+			#frame_support::RuntimeDebugNoBound,
+			#frame_support::CloneNoBound,
+			#frame_support::EqNoBound,
+			#frame_support::PartialEqNoBound,
+			#frame_support::__private::codec::Encode,
+			#frame_support::__private::codec::Decode,
+			#frame_support::__private::scale_info::TypeInfo,
+		)]
+		#[codec(encode_bound())]
+		#[codec(decode_bound())]
+		#[scale_info(skip_type_params(#type_use_gen), capture_docs = #capture_docs)]
+		pub struct #query_struct_ident<#type_decl_bounded_gen> #where_clause {
+			#(
+				pub #arg_names: #arg_types,
+			)*
+			_marker: ::core::marker::PhantomData<(#type_use_gen,)>,
+		}
+
+		impl<#type_impl_gen> #query_struct_ident<#type_use_gen> #where_clause {
+			/// Create a new [`#query_struct_ident`] instance.
```

Comment:
Nit: this outputs as written here rather than `#query_struct_ident` being substituted with the ident as you'd want.

Best bet might be to build the doc string beforehand and then use `#[doc = #doc_string]` to get it in!

```
#[doc = ""
```